docs/Hooks.md

```diff
@@ -141,6 +141,48 @@ fastify.addHook('onSend', (request, reply, payload, next) => {
 
 Note: If you change the payload, you may only change it to a `string`, a `Buffer`, a `stream`, or `null`.
 
+### Respond to a request from an hook
+If need you can respond to a request before you reach the route handler, an example could be an authentication hook. If you are using `onRequest` or a middleware just use `res.end`, if you are using the `preHandler` hook use `reply.send`. Remember to always call `next` if you are using the standard hook api, if you are working with *async* hooks it will be done automatically by Fastify.
+```js
+// standard api
+fastify.addHook('preHandler', (request, reply, next) => {
+  reply.send({ hello: 'world' })
+  next()
+})
+
+// async api
+fastify.addHook('preHandler', async (request, reply) => {
+  reply.send({ hello: 'world' })
+})
+```
+
+If you want to respond with a stream, you should make sure that you don't call `next` before the response has finished.
+
+```js
+// standard api
+fastify.addHook('onRequest', (req, res, next) => {
+  const stream = fs.createReadStream('some-file', 'utf8')
+  stream.pipe(res)
+  res.once('finish', next)
+})
+
+// standard api
+fastify.addHook('preHandler', (request, reply, next) => {
+  const stream = fs.createReadStream('some-file', 'utf8')
+  reply.send(stream)
+  reply.res.once('finish', next)
```

Comment:
We should really not be needing this.
We should not really be encouraging anyone to do this, so it's better if we __do not document this behavior__.
Doing this in their code is substantially wrong.

Reply:
So your suggestion is to remove the documentation about respond with streams inside the hooks?

Reply:
I would say so. We definitely need a better story for this hook, and there is solid work already done here which I do not want to block. Can you open a separate issue about replying with streams in a `'preHandler'`?

Reply:
Ok, I've updated the docs.